crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1187,6 +1188,24 @@ fn handle_pattern_match_as_comment<'a>(
     }
 }
 
+/// Handles dangling comments between the `*` token and identifier of a pattern match star:
+///
+/// ```python
+/// case [
+///   ...,
+///   *  # dangling end of line comment
+///   # dangling end of line comment
+///   rest,
+/// ]: ...
+/// ```
+fn handle_pattern_match_star_comment(comment: DecoratedComment) -> CommentPlacement {
+    debug_assert!(comment.enclosing_node().is_pattern_match_star());
+    let AnyNodeRef::PatternMatchStar(range, ..) = comment.enclosing_node() else {
```

Comment:
You can get the `ast::PatternMatchStar` from the `AnyNodeRef::PatternMatchStar(_)` in the match and then pass it to the function (like `handle_leading_class_with_decorators_comment`)

Reply:
Ah I see, this doesn't work.

Reply:
I think I'd suggest treating these as dangling, and then rendering them as leading comments within the node.

Reply:
(Leading is more consistent with how we treat comments in starred _expressions_.)

Reply:
(Tweaked this to merge.)